### PR TITLE
361 addWeapon update

### PR DESCRIPTION
**Pull request by NikkoJT**
Changes all instances of addWeaponCargo to addWeaponWithAttachmentsCargoGlobal for better magazine adding as mentioned in #361 

In the case of AT, preserved the same number of total AT rounds as before. In the case of carbines and SMGs, don't care.

### CODE REVIEW

**Comment by SamLex** [comment]
One concern I have with `addWeaponWithAttachmentsCargo` is that:

> If the weapon class has any pre-attached items (LinkedItems), they are removed

This might be confusing from mission makers who expect these weapons to have their "default" attachments (not sure of examples of weapons that have this but sure there will be some).

Is it possible to use the `_attachments` array and such to give these weapons the configured attachments? Might be tricky since `addWeaponWithAttachmentsCargo` wants the attachments broken into each type.

**Comment by NikkoJT** [comment]
I'll have to look at the config for prefab weapons but I'm not sure whether it is possible.

I don't necessarily consider this likely to come up because when F3 adds weapons to units through assignGear, it automatically strips them of all attachments, so that should be the expected behaviour anyway.

**Comment by NikkoJT** [comment]
It is possible to retrieve a weapon's preconfigured linkedItems from config. However, this requires a config lookup which I think is likely to be excessively complex for placement in assignGear loadouts, which are otherwise very simple for the user to edit (e.g. this would need multiple lines for this one weapon, whereas normally each line is its own thing).

Given that assignGear weapons are stripped of their linkedItems when used in a unit loadout, I think it's fair to keep that behaviour here as well, rather than making the vehicle loadouts overcomplicated. That means F3's preference for attachment handling will continue to be via the F3-defined _attach1, _silencer1 etc as with unit loadouts.

File: f/assignGear/f_assignGear_3IFB_v.sqf [hunk 1]
```diff
@@ -7,7 +7,7 @@
 		clearMagazineCargoGlobal _unit;
 		clearItemCargoGlobal _unit;
 		clearBackpackCargoGlobal _unit;
-		_unit addWeaponCargoGlobal [_carbine, 2];
+		_unit addWeaponWithAttachmentsCargoGlobal [[_carbine,"","","",[_carbinemag,500],[],""],2];
```
Comment (SamLex) [inline f/assignGear/f_assignGear_3IFB_v.sqf:10]: I'm guessing 500 is a general value to ensure it's always a full mag?

Comment (NikkoJT) [reply f/assignGear/f_assignGear_3IFB_v.sqf:10]: Yes, the command automatically trims large numbers to the maximum possible in the given magazine.

**Comment by NikkoJT** [comment]
> Although F3 does strip the attachments, it does then add user-configured ones from `_attachments` etc. So, these will be missing from weapons in cargo (but it's already the case that weapons in cargo ignore `_attachments`). It'd probably also be to complex to have the weapons in cargo use `_attachments`.

It's not sensibly possible to use the `_attachments` array itself because we can't guarantee it's in the right order - `_attachments select 0` could return a scope or a rail attachment or a bipod...and the difference matters for the command. We could use the individual `_scope1`, `_attach1` etc but it's not really safe to assume the missionmaker wants all of those to be used by default. If the missionmaker _does_ want attachments on cargo weapons, they can always plug in the attachment variables themselves.

**Review by SamLex — APPROVED** [review]
Looks good to me

**Pull request merged**